### PR TITLE
[Feat] run func timeout use std::time::Duration

### PR DESCRIPTION
**Pull request by L-jasmine**

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: crates/wasmedge-sys/src/executor.rs [hunk 1]
```diff
@@ -402,7 +402,8 @@ impl Executor {
                 )));
             }
             let mut value: libc::itimerspec = std::mem::zeroed();
-            value.it_value.tv_sec = timeout as i64;
+            value.it_value.tv_sec = timeout.as_secs() as _;
+            value.it_value.tv_nsec = timeout.subsec_nanos() as _;
```
Comment (apepkuss) [inline crates/wasmedge-sys/src/executor.rs:406]: For better readability, please specify the concrete type used by `as` operator. Thanks a lot!

File: crates/wasmedge-sys/src/async/fiber.rs [hunk 2]
```diff
@@ -115,15 +115,15 @@ impl<'a> TimeoutFiberFuture<'a> {
     ///
     /// * `func` - The function to execute.
     ///
-    /// * `timeout_sec` - The maximum execution time in seconds for the function instance.
+    /// * `deadline` - The deadline the function instance.
```
Comment (apepkuss) [inline crates/wasmedge-sys/src/async/fiber.rs:118]: Please refine the description for the `deadline` argument. Thanks a lot!

File: crates/wasmedge-sys/src/async/fiber.rs [hunk 3]
```diff
@@ -184,8 +184,16 @@ impl<'a> Future for TimeoutFiberFuture<'a> {
                 if libc::timer_create(libc::CLOCK_REALTIME, &mut sev, &mut timerid) < 0 {
                     return Poll::Ready(Err(()));
                 }
+
+                let timeout = self.deadline.duration_since(std::time::SystemTime::now());
+                if timeout.is_err() {
+                    return Poll::Ready(Err(()));
+                }
+                let timeout = timeout.unwrap().max(std::time::Duration::from_millis(100));
```
Comment (apepkuss) [inline crates/wasmedge-sys/src/async/fiber.rs:192]: Please refer to the following code to refine the code snippet for avoiding `unwrap`. Thanks!

```rust
let timeout = match self.deadline.duration_since(std::time::SystemTime::now()) {
    Ok(timeout) => timeout.max( ... ),
    Err(_) => return Poll::Ready(Err(()))
}
```

File: crates/wasmedge-sys/src/async/fiber.rs [hunk 4]
```diff
@@ -115,15 +115,15 @@ impl<'a> TimeoutFiberFuture<'a> {
...
     /// * `func` - The function to execute.
     ///
-    /// * `timeout_sec` - The maximum execution time in seconds for the function instance.
+    /// * `deadline` - The deadline the function instance.
     ///
     /// # Error
     ///
     /// If fail to create the fiber stack or the fiber fail to resume, then an error is returned.
     pub(crate) async fn on_fiber<R>(
         async_state: &AsyncState,
```
Comment (apepkuss) [inline crates/wasmedge-sys/src/async/fiber.rs:124]: Please add the description for the `async_state` argument.

**Review by apepkuss — CHANGES_REQUESTED** [review]

**Review by apepkuss — APPROVED** [review]

**Pull request merged**